src/core/sync/mutex.d

```diff
@@ -296,6 +296,11 @@ package:
         {
             return &m_hndl;
         }
+
+        shared(pthread_mutex_t*) handleAddr() shared
```

Comment:
This seems unecessary?

Reply:
Actually, does the non-`shared` `handleAddr` need to exist? It's marked as package, so I see no problem with also making it as `final nothrow @nogc`.

Reply:
If we fix this by using overloading such as `Mutex`, both versions are needed because they are called from `shared` and non-`shared` `wait()`, respectively.
If we fix it by using casting as @MoonlightSentinel suggested, only non-`shared` version is needed.

Which solution is more preferable?

Reply:
I will remove it because all the use of `handleAddr` is to be passed to `pthread_cond_*` that requires `pthread_mutex_t*`.
Casting `Condition#m_assocMutex` to `Mutex` seems better rather than providing an overload of `handleAddr` for `shared Mutex` because it is finally casted to non-`shared`.